internal/llrp/device_service.go

```diff
@@ -0,0 +1,294 @@
+//
+// Copyright (C) 2020 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package llrp
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/pkg/errors"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+// These are the names of deviceResource and deviceCommands
+// this expects the devices are using in their registered deviceProfiles.
+const (
+	basePath = "/api/v1/device/name/" // paths are {base/}{device}{/target}
+
+	capDevCmd    = "/capabilities"
+	configDevCmd = "/config"
+	addCmd       = "/roSpec"
+	enableCmd    = "/enableROSpec"
+	disableCmd   = "/disableROSpec"
+	stopCmd      = "/stopROSpec"
+	startCmd     = "/startROSpec"
+	deleteCmd    = "/deleteROSpec"
+
+	enableImpinjCmd = "/enableImpinjExt"
+
+	capReadingName = "ReaderCapabilities"
+
+	maxBody = 100 * 1024
+)
+
+// DSClient is a client to interact with the LLRP Device Service.
+type DSClient struct {
+	baseURL    string
+	httpClient *http.Client
+}
+
+// NewDSClient returns a DSClient reachable at the given host URL,
+// using the given http Client, which of course may be the default.
+func NewDSClient(host *url.URL, c *http.Client) DSClient {
```

Comment:
Add TODO to use new device service Clients from go-mod-core-clients when upgrading to V2.